clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -524,7 +525,15 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new AuthenticationFailedException(message);
             }
-        });
+    }),
+    UNKNOWN_PRODUCER(59, "The specified producerId does not exist on the broker. This may indicate data loss on the broker " +
+            "and should be investigated.",
```

Comment:
This message is more alarming than the exception Javadoc. Either we should make the other one more alarming or this one less. :)